src/System.Text.Json/tests/NewtonsoftTests/CustomObjectConverterTests.cs

```diff
@@ -158,17 +158,32 @@ public void AssertShouldSerializeTest()
         [Fact]
         public void AssertDoesNotDeserializeInterface()
         {
-            const string json = @"{
+            const string validJson = @"[{
 ""Value"": ""A value"",
 ""Thing"": {
 ""Number"": 123
 }
-}";
-            NotSupportedException e = Assert.Throws<NotSupportedException>(() =>
+]";
+            NotSupportedException nse = Assert.Throws<NotSupportedException>(() =>
             {
-                JsonSerializer.Deserialize<List<MyClass>>(json);
+                JsonSerializer.Deserialize<List<MyClass>>(validJson);
             });
-            Assert.Equal("Deserialization of interface types is not supported. Type 'System.Text.Json.Tests.IThing'", e.Message);
+            Assert.Equal("Deserialization of interface types is not supported. Type 'System.Text.Json.Tests.IThing'", nse.Message);
+
+            const string invalidJson = @"{
```

Comment:
Why is this Json invalid? Is it because it is incomplete/doesn't have the closing brace? Otherwise, the JSON seems valid.

Reply:
It's invalid for the type we are trying to deserialize into. A `List<>` should start with `[`.